### ISSUE TITLE
rustdoc-ui is failing for my local (stage1) build (was: "I cannot tell which test suites the CI will run")

### ISSUE THREAD

**Issue by pnkfelix** [rust-lang/rust#86366]
I saw some local failures when I ran `x.py test src/test/rustdoc-ui`, running against an unmodified checkout.

From this, I infer that our CI is not testing that directory, at least not on a x86_64 Linux host.

There are other test suites in a somewhat similar situation, such as `src/test/debuginfo`, where I think we are not testing against `lldb` anywhere in CI (though I do think those tests *are* exercised against `gdb`, so that's not quite as bad).

Anyway, if there is information about which test suites are exercised by the CI and which ones are not, I could not immediately find it.

And it probably would be good to have some way to double-check such things. E.g. a README in each test suite directory that includes some section that says if it not run by CI (and, maybe, for extra credit, if it is run by CI, then list the CI configs that will run that test suite. And then have the bors leave a comment when the entries go stale -- I don't think such a scenario warrants blocking PRs, but nagging comments seem reasonable).

**Comment by pnkfelix** [comment]
@jyn514 reports to me that the CI definitely is running tests against `src/test/rustdoc-ui` 

The reason I don't see `rustdoc-ui` in the CI scripts is that the run of that test suite is part of the runs that are implied by running `x.py test` with no filters on the test suite. @jyn514 advised me to find that spot in the CI scripts by grepping for "--exclude src/tools/tidy`, which pointed me here:

https://github.com/rust-lang/rust/blob/8daad743c4bb941536860b4df20111cb71b8c777/src/ci/docker/host-x86_64/x86_64-gnu-llvm-10/Dockerfile#L35-L52

So maybe my request for automatically-maintained info in a per-test-suite README is overkill. (But at the same time, it doesn't seem absurd for us to provide it.)

**Comment by jyn514** [comment]
Can you please post what failure you're seeing?

**Comment by Mark-Simulacrum** [comment]
I've long wanted us to collect and track which test suites are run on each builder, but we don't currently do that. It would not be *too* hard to start, but definitely represents a good amount of time investment on the design and implementation.

**Comment by pnkfelix** [comment]
> Can you please post what failure you're seeing?

Pasted in: https://gist.github.com/pnkfelix/99e0203960c48994c2bf93064b0344c2

Many of the failures seem to be caused by the actual output omitting the expected text "compile fail". @jyn514 [hypothesized on Zulip](https://zulip-archive.rust-lang.org/131828tcompiler/02337Testmetacoverage.html#242905871) that maybe the test suite has a bug that is causing the wrong binary to be invoked.

**Comment by pnkfelix** [comment]
bisection points to PR #84863, but @jyn514 has privately indicated that PR #84863 is most likely not itself incorrect.

Rather it is more likely that PR #84863 has exposed a latent bug in our test suite.

**Comment by ehuss** [comment]
The issue is that the default for `./x.py test` is `--stage 1`, which means that rustdoc will be built with the beta compiler.  rustdoc is using the test framework from beta (see conditional [here](https://github.com/rust-lang/rust/blob/a85f584aebd9b08314bf30b9adc17b4a752143e5/src/librustdoc/doctest.rs#L945-L948)), which does not have the same output.

I'm not sure how that is usually handled.  Maybe add `// ignore-stage1` on those tests?  However, someone will need to remember to remove those.

I'm kinda surprised more people haven't reported this, I would assume some people are running these tests locally?

**Comment by jyn514** [comment]
> I'm kinda surprised more people haven't reported this, I would assume some people are running these tests locally?

Almost everyone on the rustdoc team uses `download-rustc`, which compiles and tests stage 2. So the only people running into this are compiler devs testing rustdoc.

**Comment by ExpHP** [comment]
> I'm kinda surprised more people haven't reported this, I would assume some people are running these tests locally?

I am surprised as well!  As a data point, I ran into this yesterday naively running `./x.py test` without any further arguments. (I am a very infrequent contributor and was just doing this from memory, and I wanted to check my WSL environment before starting development).

**Comment by jyn514** [comment]
The `cfg(bootstrap)` line has since been removed, so this specific issue has been fixed. https://github.com/rust-lang/compiler-team/issues/439 seems like a decent idea but I don't think it needs its own issue in rust-lang/rust until the MCP has been accepted.
